tests/integration/suite/placement/authz/mtls.go

```diff
@@ -97,53 +97,58 @@ func (m *mtls) Run(t *testing.T, ctx context.Context) {
 	client := v1pb.NewPlacementClient(conn)
 
 	// Can only create hosts where the app ID match.
-	stream := establishStream(t, ctx, client)
-	require.NoError(t, stream.Send(&v1pb.Host{
+	// When no namespace is sent in the message, and tls is enabled
+	// the placement service will infer the namespace from the SPIFFE ID.
+	_, err = establishStream(t, ctx, client, &v1pb.Host{
 		Id: "app-1",
-	}))
-	waitForUnlock(t, stream)
-	_, err = stream.Recv()
+	})
 	require.NoError(t, err)
 
-	stream = establishStream(t, ctx, client)
-	require.NoError(t, stream.Send(&v1pb.Host{
+	_, err = establishStream(t, ctx, client, &v1pb.Host{
 		Id: "app-2",
-	}))
-	waitForUnlock(t, stream)
-	_, err = stream.Recv()
+	})
 	require.Error(t, err)
-	assert.Equal(t, codes.PermissionDenied, status.Code(err))
-}
+	require.Equal(t, codes.PermissionDenied, status.Code(err))
 
-func waitForUnlock(t *testing.T, stream v1pb.Placement_ReportDaprStatusClient) {
-	t.Helper()
-	assert.EventuallyWithT(t, func(c *assert.CollectT) {
-		resp, err := stream.Recv()
-		//nolint:testifylint
-		if assert.NoError(c, err) {
-			assert.Equal(c, "unlock", resp.GetOperation())
-		}
-	}, time.Second*5, time.Millisecond*10)
+	// Older sidecars (pre 1.4) will not send the namespace in the message.
+	// In this case the namespace is inferred from the SPIFFE ID.
+	_, err = establishStream(t, ctx, client, &v1pb.Host{
+		Id:        "app-1",
+		Namespace: "",
+	})
+	require.NoError(t, err)
+
+	// The namespace id in the message and SPIFFE ID should match
+	_, err = establishStream(t, ctx, client, &v1pb.Host{
+		Id:        "app-1",
+		Namespace: "default",
+	})
+	require.NoError(t, err)
+
+	_, err = establishStream(t, ctx, client, &v1pb.Host{
+		Id:        "app-1",
+		Namespace: "foo",
+	})
+	require.Error(t, err)
+	require.Equal(t, codes.PermissionDenied, status.Code(err))
 }
 
-func establishStream(t *testing.T, ctx context.Context, client v1pb.PlacementClient) v1pb.Placement_ReportDaprStatusClient {
+func establishStream(t *testing.T, ctx context.Context, client v1pb.PlacementClient, firstMessage *v1pb.Host) (v1pb.Placement_ReportDaprStatusClient, error) {
 	t.Helper()
 	var stream v1pb.Placement_ReportDaprStatusClient
-	assert.EventuallyWithT(t, func(c *assert.CollectT) {
-		var err error
+	var err error
+
+	require.EventuallyWithT(t, func(c *assert.CollectT) {
```

Comment:
This `Eventually` isn't doing anything.

Reply:
Ah, yes, the last line went missing during the refactor... Putting it back.

Reply:
We use it as a retry until placement is ready, and then check the error outside of the function.

Reply:
The `Eventually` is only going to be run once as `c` is not asserted. It doesn't seem like we need this `Eventually` if the tests are passing?

Reply:
Ah yes! Fixed.